### PR TITLE
JS: add support for import.meta expressions in JavaScript

### PR DESCRIPTION
**Pull request by erik-krogh**
My [earlier pull request](https://github.com/Semmle/ql/pull/2169) added support for `import.meta` expressions in TypeScript, but we never got around to supporting `import.meta` JavaScript. 

This PR adds support for `import.meta` in JavaScript, and fixes the syntax errors mentioned in [this issue](https://github.com/Semmle/ql/issues/2158). 

No db upgrade is required, as the previous TypeScript PR already added most of the required code.

### CODE REVIEW

File: javascript/extractor/src/com/semmle/jcorn/ESNextParser.java [hunk 1]
```diff
@@ -240,6 +240,9 @@ protected Expression parseExprAtom(DestructuringErrors refDestructuringErrors) {
     if (this.type == TokenType._import) {
       Position startLoc = this.startLoc;
       this.next();
+      if (this.type == TokenType.dot) {
```
Comment (max-schaefer) [inline javascript/extractor/src/com/semmle/jcorn/ESNextParser.java]: You could use `this.eat(TokenType.dot)` here, then you don't need the `this.next()` on line 424 below.

File: javascript/extractor/src/com/semmle/jcorn/ESNextParser.java [hunk 2]
```diff
@@ -413,6 +416,20 @@ protected Statement parseImport(Position startLoc) {
     }
   }
 
+  /**
+   * Parses an import.meta expression, assuming that the initial "import" has been consumed.
+   */
+  private MetaProperty parseImportMeta(Position loc) {
+    Identifier meta = new Identifier(new SourceLocation(loc), "import");
```
Comment (max-schaefer) [inline javascript/extractor/src/com/semmle/jcorn/ESNextParser.java]: Not sure "meta" is a great name; it's the property that's `meta`, not the `import`.

Comment (erik-krogh) [reply javascript/extractor/src/com/semmle/jcorn/ESNextParser.java]: It follows the naming in `MetaProperty`.  
The "meta"-object is `import`, and the property on the "meta"-object is `meta`. 

But yes, its an unfortunate clash of names, so I'm renaming it to `importIdentifier`.

Comment (erik-krogh) [reply javascript/extractor/src/com/semmle/jcorn/ESNextParser.java]: Actually no. I'll inline the creation of the identifier.

File: change-notes/1.24/analysis-javascript.md [hunk 3]
```diff
@@ -27,4 +27,4 @@
 ## Changes to libraries
 
 * The predicates `RegExpTerm.getSuccessor` and `RegExpTerm.getPredecessor` have been changed to reflect textual, not operational, matching order. This only makes a difference in lookbehind assertions, which are operationally matched backwards. Previously, `getSuccessor` would mimick this, so in an assertion `(?<=ab)` the term `b` would be considered the predecessor, not the successor, of `a`. Textually, however, `a` is still matched before `b`, and this is the order we now follow.
-
+* `import.meta` expressions no longer result in a syntax error in JavaScript files.
```
Comment (max-schaefer) [inline change-notes/1.24/analysis-javascript.md]: This should go into `extractor-javascript.md`

**Pull request merged**